uninstall/action.yml

```diff
@@ -40,37 +47,60 @@ runs:
         BRANCH_FILTER: ${{ inputs.branch_filter }}
+        BRANCH_HELM_PROPERTY: ${{ inputs.branch_helm_property }}
+        MAX_AGE_IN_DAYS: ${{ inputs.max_age_in_days }}
       run: |
         set -eu
         HELM_ARGS=$(cat /tmp/HELM_ARGS)
 
         echo "Filtered helm releases by name:"
-        HELM_RELEASES=$(helm list -q $HELM_ARGS -n "${NAMESPACE}" -f "${RELEASE_FILTER}")
+        HELM_RELEASES_JSON=$(helm list $HELM_ARGS -n "${NAMESPACE}" -o json | jq -r --arg REG ${RELEASE_FILTER} '.[] | select(.name|test($REG))' | jq -s '.')
+        HELM_RELEASES=$(echo "${HELM_RELEASES_JSON}" | jq -r '.[].name')
+        echo -n "${HELM_RELEASES}" > /tmp/HELM_RELEASES
+
         if [ -n "${HELM_RELEASES}" ]; then
           echo "${HELM_RELEASES}"
         fi
         echo
 
-        if [ -z "${BRANCH_FILTER}" ]; then
-          echo -n "${HELM_RELEASES}" > /tmp/HELM_RELEASES
-          exit 0
+        if [ ! -z "${BRANCH_FILTER}" ] && [ -n "${HELM_RELEASES}" ]; then
+          echo "Filtered helm releases by branch $BRANCH_FILTER:"
+          echo -n > /tmp/HELM_RELEASES
+          for RELEASE in $HELM_RELEASES
+          do
+            set +e
+            HELM_VALUES_JSON=$(helm get values $HELM_ARGS -n "${NAMESPACE}" --all -o json "${RELEASE}")
+            BRANCH=$(echo "${HELM_VALUES_JSON}" | jq -r --arg BRANCH_HELM_PROPERTY "${BRANCH_HELM_PROPERTY}" '$BRANCH_HELM_PROPERTY')
+            set -e
+
+            if [ "$BRANCH" == "$BRANCH_FILTER" ]; then
+              echo "$RELEASE"
+              echo -e "$RELEASE" >> /tmp/HELM_RELEASES
+            fi
+          done
+          echo
         fi
 
-        echo "Filtered helm releases by branch $BRANCH_FILTER:"
-        echo -n > /tmp/HELM_RELEASES
-        for RELEASE in $HELM_RELEASES
-        do
-          set +e
-          HELM_VALUES_JSON=$(helm get values $HELM_ARGS -n "${NAMESPACE}" --all -o json "${RELEASE}")
-          BRANCH=$(echo "${HELM_VALUES_JSON}" | jq -r '.branch' )
-          set -e
+        HELM_RELEASES=$(cat /tmp/HELM_RELEASES)
 
-          if [ "$BRANCH" == "$BRANCH_FILTER" ];	then
-            echo "$RELEASE"
-            echo -e "$RELEASE" >> /tmp/HELM_RELEASES
-          fi
-        done
-        echo
+        if [ ! -z "${MAX_AGE_IN_DAYS}" ] && [ -n "${HELM_RELEASES}" ]; then
+          echo "Filtered helm releases by date:"
+          echo -n > /tmp/HELM_RELEASES
+          for RELEASE in $HELM_RELEASES
+          do
+            LAST_UPGRADE=$(echo "${HELM_RELEASES_JSON}" | jq -r ". | map(select(.name == \"$RELEASE\"))[0].updated" | cut -d ' ' -f1-2)
+            LAST_UPGRADE_DATE=$(date -d "${LAST_UPGRADE}" +%s)
+            COMPARE_DATE=$(date -d "now - ${MAX_AGE_IN_DAYS} days" +%s)
+
+            if [ "${LAST_UPGRADE_DATE}" -le "${COMPARE_DATE}" ]; then
+              echo "$RELEASE"
+              echo -e "$RELEASE" >> /tmp/HELM_RELEASES
+            fi
+          done
+        fi
+
+        echo -n "${HELM_RELEASES}" > /tmp/HELM_RELEASES
```

Comment:
this will  overwrite your  releases filtered by max-age, so this filter will never work